keystores/src/main/java/com/palantir/conjure/java/config/ssl/SslSocketFactories.java

```diff
@@ -91,6 +92,20 @@ public static SSLSocketFactory createSslSocketFactory(
         return sslContext.getSocketFactory();
     }
 
+    /**
+     * Create a {@link SSLSocketFactory} from the provided certificates and configuration.
+     *
+     * @param trustCertificatesByAlias a map of X.509 certificate in PEM or DER format by the alias to load the
+     *     certificate as.
+     * @param config an {@link SslConfiguration} describing the trust store configuration
+     * @param provider The preferred security {@link Provider}
+     */
+    public static SSLSocketFactory createSslSocketFactory(
```

Comment:
I noticed most of these methods aren't explicitly tested, do I need to add tests for my additions?